src/renderer/components/dock/dock-tabs.tsx

```diff
@@ -41,21 +36,22 @@ interface Props {
 }
 
 export const DockTabs = ({ tabs, autoFocus, selectedTab, onChangeTab }: Props) => {
-  const renderTab = (tab: IDockTab) => {
-    if (isTerminalTab(tab)) {
-      return <TerminalTab value={tab} />;
+  const renderTab = (tab?: IDockTab) => {
+    if (!tab) {
+      return null;
     }
 
-    if (isCreateResourceTab(tab) || isEditResourceTab(tab)) {
-      return <DockTab value={tab} icon="edit" />;
-    }
-
-    if (isInstallChartTab(tab) || isUpgradeChartTab(tab)) {
-      return <DockTab value={tab} icon={<Icon svg="install" />} />;
-    }
-
-    if (isLogsTab(tab)) {
-      return <DockTab value={tab} icon="subject" />;
+    switch (tab.kind) {
+      case TabKind.CREATE_RESOURCE:
+      case TabKind.EDIT_RESOURCE:
+        return <DockTab value={tab} icon="edit" />;
+      case TabKind.INSTALL_CHART:
+      case TabKind.UPGRADE_CHART:
+        return <DockTab value={tab} icon={<Icon svg="install" />} />;
+      case TabKind.POD_LOGS:
+        return <DockTab value={tab} icon="subject" />;
+      case TabKind.TERMINAL:
+        return <TerminalTab value={tab} />;
     }
```

Comment:
Because we are limiting it to an `enum` then typescript considers this fully covered. Either way it will fall through, return `undefined` and then go on to not render anything.

So I think it is fine.